homeassistant/components/electric_kiwi/config_flow.py

```diff
@@ -40,12 +47,30 @@ async def async_step_reauth_confirm(
     ) -> ConfigFlowResult:
         """Dialog that informs the user that reauth is required."""
         if user_input is None:
-            return self.async_show_form(step_id="reauth_confirm")
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                description_placeholders={CONF_NAME: self._get_reauth_entry().title},
+            )
         return await self.async_step_user()
 
     async def async_oauth_create_entry(self, data: dict) -> ConfigFlowResult:
         """Create an entry for Electric Kiwi."""
-        existing_entry = await self.async_set_unique_id(DOMAIN)
-        if existing_entry:
-            return self.async_update_reload_and_abort(existing_entry, data=data)
-        return await super().async_oauth_create_entry(data)
+        ek_api = ElectricKiwiApi(
+            api.ConfigFlowElectricKiwiAuth(self.hass, data["token"]["access_token"])
+        )
+
+        try:
+            session = await ek_api.get_active_session()
+            unique_id = str(session.data.customer_number)
+
+            await self.async_set_unique_id(unique_id)
+            if self.source == SOURCE_REAUTH:
+                self._abort_if_unique_id_mismatch(reason="wrong_account")
+                return self.async_update_reload_and_abort(
+                    self._get_reauth_entry(), data=data
+                )
+
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=unique_id, data=data)
+        except ApiException:
```

Comment:
Can we only do the things that raise in the try block?

Reply:
open to suggestions as the unique id is pulled from the get_active_session which is used right up to async_create_entry

Reply:
Done, so weird you can assign in a try block and it flows on outside of it.